arwen/ethapi/ethereumei.go

```diff
@@ -0,0 +1,543 @@
+package ethapi
+
+// // Declare the function signatures (see [cgo](https://golang.org/cmd/cgo/)).
+//
+// #include <stdlib.h>
+// typedef unsigned char u8;
+// typedef int i32;
+// typedef int i32ptr;
+// extern void useGas(void *context, long long  gas);
+// extern void getAddress(void *context, i32ptr resultOffset);
+// extern void getExternalBalance(void *context, i32ptr addressOffset, i32ptr resultOffset);
+// extern i32 getBlockHash(void *context, long long number, i32ptr resultOffset);
+// extern i32 call(void *context, long long gas, i32ptr addressOffset, i32ptr valueOffset, i32ptr dataOffset, i32 dataLength);
+// extern i32 getCallDataSize(void *context);
+// extern void callDataCopy(void *context, i32ptr resultsOffset, i32ptr dataOffset, i32 length);
+// extern i32 callCode(void *context, long long gas, i32ptr addressOffset, i32ptr valueOffset, i32ptr dataOffset, i32 dataLength);
+// extern i32 callDelegate(void *context, long long gas, i32ptr addressOffset, i32ptr dataOffset, i32 dataLength);
+// extern i32 callStatic(void *context, long long gas, i32ptr addressOffset, i32ptr dataOffset, i32 dataLength);
+// extern void storageStore(void *context, i32ptr pathOffset, i32ptr valueOffset);
+// extern void storageLoad(void *context, i32ptr pathOffset, i32ptr resultOffset);
+// extern void getCaller(void *context, i32ptr resultOffset);
+// extern void getCallValue(void *context, i32ptr resultOffset);
+// extern void codeCopy(void *context, i32ptr resultOffset, i32 codeOffset, i32 length);
+// extern i32 getCodeSize(void *context);
+// extern void getBlockCoinbase(void *context, i32ptr resultOffset);
+// extern i32 create(void *context, i32ptr valueoffset, i32ptr dataOffset, i32 length, i32ptr resultsOffset);
+// extern void getBlockDifficulty(void *context, i32ptr resultOffset);
+// extern void externalCodeCopy(void *context, i32ptr addressOffset, i32ptr resultOffset, i32 codeOffset, i32 length);
+// extern i32 getExternalCodeSize(void *context, i32ptr addressOffset);
+// extern long long getGasLeft(void *context);
+// extern long long getBlockGasLimit(void *context);
+// extern void getTxGasPrice(void *context, i32ptr valueOffset);
+// extern void logTopics(void *context, i32ptr dataOffset, i32 length, i32 numberOftopics, i32ptr topic1, i32ptr topic2, i32ptr topic3, i32ptr topic4);
+// extern long long getBlockNumber(void *context);
+// extern void getTxOrigin(void *context, i32ptr resultOffset);
+// extern void finish(void *context, i32ptr dataOffset, i32 length);
+// extern void revert(void *context, i32ptr dataOffset, i32 length);
+// extern i32 getReturnDataSize(void *context);
+// extern void returnDataCopy(void *context, i32ptr resultOffset, i32 dataOffset, i32 length);
+// extern void selfDestruct(void *context, i32ptr addressOffset);
+// extern long long getBlockTimestamp(void *context);
```

Comment:
Some of these CGO exports seem to be conflicting with the ones from ElrondEI. This branch does not build on my machine:
```
multiple definition of `getExternalBalance' ... (etc)
multiple definition of `storageStore' ... (etc)
multiple definition of `storageLoad' ... (etc)
multiple definition of `getCaller' ... (etc)
multiple definition of `getGasLeft' ... (etc)
multiple definition of `getBlockTimestamp' ... (etc)
```

Reply:
done